universal-application-tool-0.0.1/app/views/dev/FileUploadView.java

```diff
@@ -75,10 +95,14 @@ private ContainerTag renderFiles(ImmutableList<StoredFile> files) {
   }
 
   private String getPresignedURL(StoredFile file) {
-    return s3Client.getPresignedUrl(file.getName()).toString();
+    String url = storageClient.getPresignedUrl(file.getName()).toString();
+    if (url.contains("azurite")) {
+      return url.replace("azurite", "localhost");
+    }
```

Comment:
suggestion (blocking): this looks like an Azure-specific implementation detail leaking into code that should be generic. Can this logic be pushed back into the storage client?

Reply:
moved this logic to BlobStorage.java in the AzuriteClient class